endpoints/events/event.go

```diff
@@ -0,0 +1,324 @@
+		Request: eventRequest,
+		Account: account,
+	})
+
+	// OK
+	w.WriteHeader(http.StatusOK)
+	// Add tracking pixel if format == image
+	if eventRequest.Format == analytics.Image {
+		w.Header().Add("Content-Type", e.TrackingPixel.ContentType)
+		w.Write(e.TrackingPixel.Content)
+	}
+}
+
+// Converts an EventRequest to an URL
+func EventRequestToUrl(externalUrl string, request *analytics.EventRequest) string {
+	s := fmt.Sprintf(TemplateUrl, externalUrl, request.Type, request.Bidid, request.AccountID)
+
+	return s + optionalParameters(request)
+}
+
+// Parses an EventRequest from an Http request
+func ParseEventRequest(r *http.Request) (*analytics.EventRequest, []error) {
+	event := &analytics.EventRequest{}
+	var errs []error
+	// validate type
+	if err := validateType(event, r); err != nil {
+		errs = append(errs, err)
+	}
+
+	// validate bidid
+	if bidid, err := validateRequiredParameter(r, BidIdParameter); err != nil {
+		errs = append(errs, err)
+	} else {
+		event.Bidid = bidid
+	}
+
+	// validate timestamp (optional)
+	if err := validateTimestamp(event, r); err != nil {
+		errs = append(errs, err)
+	}
+
+	// validate format (optional)
+	if err := validateFormat(event, r); err != nil {
+		errs = append(errs, err)
+	}
+
+	// validate analytics (optional)
+	if err := validateAnalytics(event, r); err != nil {
+		errs = append(errs, err)
+	}
+
+	// Bidder
+	bidder := r.FormValue(BidderParameter)
+
+	if bidder != "" {
+		event.Bidder = bidder
+	}
+
+	return event, errs
+}
+
+// Handle fetching account errors
+func HandleAccountServiceErrors(errs []error) (status int, messages []string) {
+	messages = []string{}
+	status = http.StatusInternalServerError
+
+	for _, er := range errs {
+		errCode := errortypes.ReadCode(er)
+
+		switch errCode {
+		case errortypes.BlacklistedAppErrorCode, errortypes.BlacklistedAcctErrorCode:
+			errCode = http.StatusServiceUnavailable
+		case errortypes.AcctRequiredErrorCode:
+			errCode = http.StatusBadRequest
+		}
+
+		messages = append(messages, er.Error())
+	}
+
+	return status, messages
+}
+
+func optionalParameters(request *analytics.EventRequest) string {
+	r := url.Values{}
+
+	// timestamp
+	if request.Timestamp > 0 {
+		r.Add(TimestampParameter, strconv.FormatInt(request.Timestamp, 10))
+	}
+
+	// bidder
+	if request.Bidder != "" {
+		r.Add(BidderParameter, request.Bidder)
+	}
+
+	// format
+	switch request.Format {
+	case analytics.Blank:
+		r.Add(FormatParameter, string(analytics.Blank))
+	case analytics.Image:
+		r.Add(FormatParameter, string(analytics.Image))
+	}
+
+	//analytics
+	switch request.Analytics {
+	case analytics.Enabled:
+		r.Add(AnalyticsParameter, string(analytics.Enabled))
+	case analytics.Disabled:
+		r.Add(AnalyticsParameter, string(analytics.Disabled))
+	}
+
+	opt := r.Encode()
+
+	if opt != "" {
+		return "&" + opt
+	}
+
+	return opt
+}
+
+// validate type
+func validateType(er *analytics.EventRequest, httpRequest *http.Request) error {
```

Comment:
Nitpick: The validateXXX methods all have a side affect of updating the `EventRequest`. To me, that seems like it's primary role and it will return an error only if that fails. Consider renaming to `setType`, `setFormat`, etc.. so the setting side affect is obvious - or perhaps something like `readType` which adds the connotation of reading from the request.

This is admittedly nitpick territory. Please take this as a suggestion, not a requirement.